lib/index.js

```diff
@@ -356,9 +352,131 @@ function filterDiary() {
   }
 }
 
-const redirectNewFood = () =>{
+const redirectNewFood = () => {
   document.location.href = 'http://localhost:8080/foods.html';
 }
+
+function getCheckBoxIds() {
+  event.preventDefault();
+  let diaries = document.getElementsByClassName('diary')
+  let meal = this.innerText;
+  let checkedFoods = getCheckedFoods();
+  checkedFoods.forEach(function(id) {
+    for (var i = 0; i < diaries.length; i++) { 
+      if (diaries[i].dataset.foodId === id) { 
+        addFoodToMeal(diaries[i], meal) 
+      } 
+    }
+  })
+}
+
+const getCheckedFoods = () => {
+  let checkboxes = $('input[type=checkbox]')
+  let checkedFoods = []
+  for (var i = 0; i < checkboxes.length; i++) {
+    if (checkboxes[i].checked) {
+      checkedFoods.push(checkboxes[i].dataset.checkboxId)
+    }
+  }
+  return checkedFoods
+}
+
+const addFoodToMeal = (food, meal) => {
+  let foodId = food.dataset.foodId
+  let name = food.getElementsByTagName('p')[0].innerText
+  let calories = food.getElementsByTagName('p')[1].innerText
+  let data = {name: name, calories: calories}
+  var mealId = getMealId(meal);
+  return fetch(`https://evening-peak-25213.herokuapp.com/api/v1/meals/` + mealId + `/foods/` + foodId, {
+    method: 'post',
+    headers:
+    { 'Content-Type': 'application/json' },
+    body: JSON.stringify(data)
+  })
+  .then(response => response.json())
+  // .then(getMeals)
+  // .then(clearCheckBoxes)
+  // this still requires a refresh and does not clear the checkboxes
+  .catch(error => console.log({ error }))
+};
+
+const getMealId = (meal) => {
+  if (meal === "Breakfast") {
+    return "1";
+  }
+  else if (meal === "Snacks") {
+    return "2";
+  }
+  else if (meal === "Lunch") {
+    return "3";
+  }
+  else if (meal === "Dinner") {
+    return "4";
+  }
+}
+
+const deleteBreakfastMeal = (event) => {
```

Comment:
Let's see for a future refactor if we can create one method and have this be a little more dynamic